services/job_runner_test.go

```diff
@@ -213,6 +213,25 @@ func TestJobRunner_ExecuteRun(t *testing.T) {
 	}
 }
 
+func TestExecuteRun_ExecuteRunAtBlock_savesOverridesOnError(t *testing.T) {
+	t.Parallel()
+
+	store, cleanup := cltest.NewStore()
+	defer cleanup()
+
+	job, initr := cltest.NewJobWithLogInitiator()
+	job.Tasks = []models.TaskSpec{}
+	run := job.NewRun(initr)
+
+	initialData := models.JSON{Result: gjson.Parse(`{"key":"shouldBeHere"}`)}
+	overrides := models.RunResult{Data: initialData}
+	run, err := services.ExecuteRunAtBlock(run, store, overrides, nil)
+	assert.Error(t, err)
+
+	store.One("ID", run.ID, &run)
+	assert.Equal(t, initialData, run.Overrides.Data)
```

Comment:
Do you think we should also add the negative assertion that it doesn't save overrides on success?

Reply:
I am under the impression we want to save the overrides on success or failure. I could assert for it's presence in the successful scenario as well.